panic in test framework echo

... causes mis-reporting/failed tests: https://prow.k8s.io/view/gcs/istio-prow/logs/istio-integ-k8s-tests-master/136
```
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x19ab4d2]
goroutine 338 [running]:
testing.tRunner.func1(0xc4202ac2d0)
/usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x1baa0e0, 0x2e46c90)
/usr/local/go/src/runtime/panic.go:502 +0x229
istio.io/istio/tests/integration/echo.TestEcho.func1.4.1(0x2082c40, 0xc42066e040)
/home/prow/go/src/istio.io/istio/tests/integration/echo/echo_test.go:152 +0x172
istio.io/istio/pkg/test/framework.(*Test).doRun(0xc42057e060, 0xc42066e040, 0xc4202143f0)
/home/prow/go/src/istio.io/istio/pkg/test/framework/test.go:112 +0x200
istio.io/istio/pkg/test/framework.(*Test).Run.func1(0xc4202ac2d0)
/home/prow/go/src/istio.io/istio/pkg/test/framework/test.go:87 +0x80
testing.tRunner(0xc4202ac2d0, 0xc4206fc040)
/usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
/usr/local/go/src/testing/testing.go:824 +0x2e0
FAIL	istio.io/istio/tests/integration/echo	151.795s
```
/assign @nmittler 
/assign @ozevren

--- a/pkg/test/framework/components/echo/echoboot/echoboot.go
+++ b/pkg/test/framework/components/echo/echoboot/echoboot.go
@@ -41,6 +41,7 @@ func NewBuilder(ctx resource.Context) (b echo.Builder, err error) {
 
 // NewBuilder for Echo Instances.
 func NewBuilderOrFail(t test.Failer, ctx resource.Context) echo.Builder {
+	t.Helper()
 	b, err := NewBuilder(ctx)
 	if err != nil {
 		t.Fatalf("echo.NewBuilderOrFail: %v", err)

--- a/pkg/test/framework/components/echo/kube/builder.go
+++ b/pkg/test/framework/components/echo/kube/builder.go
@@ -69,6 +69,7 @@ func (b *builder) Build() error {
 }
 
 func (b *builder) BuildOrFail(t test.Failer) {
+	t.Helper()
 	if err := b.Build(); err != nil {
 		t.Fatal(err)
 	}

--- a/pkg/test/framework/testcontext.go
+++ b/pkg/test/framework/testcontext.go
@@ -251,7 +251,7 @@ func (c *testContext) Failed() bool {
 
 func (c *testContext) Fatal(args ...interface{}) {
 	c.Helper()
-	c.T.Error(args...)
+	c.T.Fatal(args...)
 }
 
 func (c *testContext) Fatalf(format string, args ...interface{}) {

